### PR TITLE
return data to client via channel

### PR DESCRIPTION
**Pull request by Penglq**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/mysql/conn.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"crypto/x509"
 	"flag"
 	"fmt"
+	"github.com/arana-db/arana/pkg/proto"
```
Comment (jjeffcaii) [inline pkg/mysql/conn.go:26]: group imports

File: pkg/mysql/server.go [hunk 2]
```diff
@@ -1192,21 +1253,21 @@ func val2MySQL(v *proto.Value) ([]byte, error) {
 	case mysql.FieldTypeNULL:
 		// no-op
 	case mysql.FieldTypeTiny:
-		val, err := strconv.ParseInt(fmt.Sprintf("%s", v.Val), 10, 8)
+		val, err := strconv.ParseInt(fmt.Sprintf("%d", v.Val), 10, 8)
```
Comment (jjeffcaii) [inline pkg/mysql/server.go:1256]: How about using `fmt.Sprint(v.Val)` directly?

File: pkg/proto/runtime.go [hunk 3]
```diff
@@ -67,6 +67,8 @@ type (
 
 	// Callable represents sql caller.
 	Callable interface {
+		// CallToIter executes a sql, return a iterator
+		CallToIter(ctx context.Context, sql string, args ...interface{}) (res Result, warn uint16, err error)
```
Comment (jjeffcaii) [inline pkg/proto/runtime.go:71]: Overwrite the `Call` function directly.

**Review by jjeffcaii — APPROVED** [review]

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/plan/union.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ package plan
 
 import (
 	"context"
+	"github.com/arana-db/arana/pkg/mysql"
```
Comment (AlexStocks) [inline pkg/runtime/plan/union.go:22]: split

File: pkg/runtime/optimize/optimizer_test.go [hunk 5]
```diff
@@ -47,7 +47,13 @@ func TestOptimizer_OptimizeSelect(t *testing.T) {
 	conn.EXPECT().Query(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
 		DoAndReturn(func(ctx context.Context, db string, sql string, args ...interface{}) (proto.Result, error) {
 			t.Logf("fake query: db=%s, sql=%s, args=%v\n", db, sql, args)
-			return nil, nil
+			return &mysql.Result{
```
Comment (AlexStocks) [inline pkg/runtime/optimize/optimizer_test.go:50]: 这里面的 field 全是默认值，没必要在这里赋值

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**